haskell-indentation.el

```diff
@@ -323,7 +331,31 @@ fixes up only indentation."
            (car (haskell-indentation-first-indentation)) cursor-in-whitespace)
         (haskell-indentation-reindent-to pi cursor-in-whitespace))))))
 
-
+(defun haskell-indentation-common-electric-command (arg)
+  "Call `self-insert-command' to insert the character typed ARG times
+and indent when all of the following are true:
+
+1) The character is the first non-whitespace character on the line.
+2) There is only one possible indentation position.
+3) The variable `electric-indent-mode' or `haskell-indentation-electric-flag'
+   is non-nil.
+4) The point is not in a comment, string, or quasiquote."
+  (interactive "*p")
+  (let* ((col (haskell-indentation-current-indentation))
```

Comment:
Also: please check the `(or haskell-indentation-electric-flag electric-indent-mode)` in the beginning of the function. The point is that `haskell-indentation-current-indentation` may possibly be buggy, take lots of time or error-out. Therefore it should be invoked only after flag is enabled.